pandas/core/sparse/array.py

```diff
@@ -595,12 +595,13 @@ def fillna(self, value, downcast=None):
         if issubclass(self.dtype.type, np.floating):
             value = float(value)
 
+        new_values = self.sp_values.copy()
+        new_values[isnull(new_values)] = value
+
         if self._null_fill_value:
-            return self._simple_new(self.sp_values, self.sp_index,
```

Comment:
I would make the if/then much simpler then (IOW you are now just choosing the fill value. also be explicit and use ``isnull(value)``